pkg/agent/route/route_linux.go

```diff
@@ -277,12 +277,24 @@ func (c *Client) syncRoute() error {
 		})
 	}
 	if c.nodeConfig.PodIPv6CIDR != nil {
+		// The system auto-generated IPv6 link-local route always uses "fe80::/64"
+		// as the destination no matter of the interface's global address's mask.
+		// Here we assume the IPv6 link-local address always exists on antrea-gw0
+		// to avoid unexpected issues in the IPv6 forwarding.
+		_, llrCIDR, _ := net.ParseCIDR("fe80::/64")
 		gwAutoconfRoutes = append(gwAutoconfRoutes, &netlink.Route{
 			LinkIndex: c.nodeConfig.GatewayConfig.LinkIndex,
 			Dst:       c.nodeConfig.PodIPv6CIDR,
 			Src:       c.nodeConfig.GatewayConfig.IPv6,
 			Scope:     netlink.SCOPE_LINK,
-		})
+		},
+			// Restore the IPv6 link-local route.
+			&netlink.Route{
+				LinkIndex: c.nodeConfig.GatewayConfig.LinkIndex,
+				Dst:       llrCIDR,
+				Scope:     netlink.SCOPE_LINK,
+			},
```

Comment:
Just to confirm, this won't override the system generated route if the latter exists, right?

Reply:
Having checked the system auto-generated route, it is using the same dst, src (nil), and scope as this route. So I don't think it introudces issues even if it replaced the system auto-generated entry.

Reply:
could you confirm if it overrides the system generated route if the latter exists?

Reply:
Thanks for catching it, my test result shows that it won't override the latter routes, instead, it may add a new route because of the different metrics, e.g.,
```
# ip -6 route list dev genev_sys_6081
fe80::/64 proto kernel metric 256 pref medium
fe80::/64 metric 1024 pref medium
```
After checking the code, our expectation is to not install such route if it already exists, as there is a precheck on if route equals to the existing entry before executing the `RouteReplace` operation,
```
	restoreRoute := func(route *netlink.Route) bool {
		r, ok := routeMap[route.Dst.String()]
		if ok && routeEqual(route, r) {
			return true
		}
		if err := c.netlink.RouteReplace(route); err != nil {
			klog.ErrorS(err, "Failed to sync route", "Route", route)
			return false
		}
		return true
	}
```
However, many route entries may exist on a Node with the same link-local IPv6 route on different net interfaces. And the key of the `routeMap` only contains route destination,  so function `routeEqual` may return false if the final cached route is not the one configured on antrea-gw0.

In my latest change, I update the key of `routeMap` using "$dst_$linkIndex", then we cound ensure to not re-install the link-local route if it exists.